lua/replayer_lua.cpp

```diff
@@ -64,10 +73,26 @@ extern "C" int loadReplayer(lua_State* L) {
 extern "C" int replayerSave(lua_State* L) {
   auto r = checkReplayer(L);
   auto path = luaL_checkstring(L, 2);
-  std::ofstream out(path);
-  luaL_argcheck(L, out, 2, "invalid save path");
-  out << *r;
-  out.close();
+  bool compressed = false;
+#ifdef WITH_ZSTD
+  if (lua_gettop(L) > 2) {
+    compressed = lua_toboolean(L, 3);
+  }
+#endif
+
+  if (compressed) {
+#ifdef WITH_ZSTD
```

Comment:
I don't think so, at least it seems not very Lua-like to me to raise an error when passed an additional argument that you're not supporting. What about printing a warning to stderr?